src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,495 @@
+from __future__ import annotations
+
+import pytest
+import os
+
+from lib.sssd.roles.client import Client
+from lib.sssd.roles.ipa import IPA
+from lib.sssd.roles.ldap import LDAP
+from lib.sssd.topology import KnownTopology
+
+
+@pytest.mark.tier(0)
+@pytest.mark.topology(KnownTopology.Client)
+def test_passkey__sssctl_register(client: Client):
+    """
+    :title: Register a key with sssctl
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use sssctl to register a FIDO2 key.
+        2. Check umockdev-run that the output contains the user key mapping data.
+    :expectedresults:
+        1. New key is registered
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    tc = "sssctl_reg_test1"
+    cmd = "sssctl passkey-exec --register --username=user1 --domain=ldap.test"
+
+    for record_file in ['yk.umockdev', f'{tc}.ioctl', f'{tc}.script', f'{tc}.output']:
+        src_file = f"{os.getcwd()}/data/passkey/{tc}/{record_file}"
+        client.fs.upload(src_file, f"/tmp/{record_file}")
+
+    output = client.auth.passkey.register(tc, cmd, 123456)
+    output_file = client.fs.read(f"/tmp/{tc}.output").strip()
+
+    assert output.rc == 0 and output.stdout_lines[-1].strip() == output_file
+
+
+@pytest.mark.tier(0)
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__sssctl_register_ipa(client: Client, ipa: IPA):
+    """
+    :title: Register a key with IPA sssctl command
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use sssctl to register a FIDO2 key using ipa command.
+        2. Check umockdev-run that the output contains the user key mapping data.
+    :expectedresults:
+        1. New key is registered with IPA command.
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    client.host.ssh.run('kinit admin@IPA.TEST', input='Secret123', raise_on_error=False)
```

Comment:
Is this needed?

Reply:
If it's needed then I'm fine with keeping it, but as Scott suggested we could replace by a call to a kinit function.

Reply:
I tried the client.setup() but won't work for the client, but worked for ipa. setup() to perform the kinit?

Reply:
@pbrezina can you help me here?

Reply:
It would be better to avoid calling this on the client but rather call it on ipa. Add method for this to the IPA role and you do not have to call expect, just provide the pin in stdin of the command.

Reply:
@madhuriupadhye could you run a quick check to make sure the umockdev files will work if your running it from the ipa server?  I don't see why they wouldn't but, it does mean that now we also have to transfer the files to the ipa server and not just the client.  But, you do that on a case by case basis so it shouldn't be that bad.

Reply:
This one is resolved now as the register is done on the IPA server in the updated test.